arangod/Aql/QueryContext.cpp

```diff
@@ -51,10 +51,10 @@ using namespace arangodb;
 using namespace arangodb::aql;
 
 /// @brief creates a query
-QueryContext::QueryContext(TRI_vocbase_t& vocbase)
+QueryContext::QueryContext(TRI_vocbase_t& vocbase, QueryId id)
     : _resourceMonitor(GlobalResourceMonitor::instance()),
       _baseOverHeadTracker(_resourceMonitor, baseMemoryUsage),
-      _queryId(TRI_NewServerSpecificTick()),
+      _queryId(id ? id : TRI_NewServerSpecificTick()),
```

Comment:
```suggestion
      _queryId(id != 0 ? id : TRI_NewServerSpecificTick()),
```
I like tests on numbers to be a bit more specific, we had a lot of bool casting errors in the past.